### DISCUSSION TITLE
add comand delay in scheduling on change/condition

### DISCUSSION THREAD

**Discussion by johkn** [emsesp/EMS-ESP32#2234] [Q&A]
example of an automation in HA

If boiler_heating_active = on
delay xy sec/min
then set boilhystoff = value

my question: would it be possible to add a delay function?

**Comment by proddy** [comment]
hmm, been thinking how to do this. It ain't easy!

**Comment by MichaelDvP** [comment]
For on_change i have an idea for implementation, but if there is a second change within the delay it get's chaotic and does not work.
For condition-trigger it's really hard to implement because we want it single action and only reactivate on condition change.

**Comment by proddy** [comment]
I was thinking of an on_change with a `time+n` in the `condition` that calls another Immediate scheduler command as the `command`

**Comment by proddy** [comment]
@johkn Since you're using Home Assistant, why not do the logic there in a trigger? It'll be so much easier, and it's also what HA is designed to do.

**Comment by johkn** [comment]
@proddy Yes, I have home assistant and I control it via automation. If the network or server is defective then the control doesn't work either. Hence the request. I want the heating to run independently and not be dependent on Smarthome.

Current automation: 

boiler_heating_active = off -> boilhystoff = 10 

boiler_heating_active = on -> delay 60 sec -> boilhystoff = 2
